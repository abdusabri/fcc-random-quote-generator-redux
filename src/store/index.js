import { createStore, applyMiddleware } from "redux";
import { appReducer } from "../reducers/combined-reducers";
import initialState from "./initial-state";
import thunk from "redux-thunk";

export default createStore(appReducer, initialState, applyMiddleware(thunk));
