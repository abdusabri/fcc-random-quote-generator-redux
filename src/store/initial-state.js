export default {
  quote: {
    quoteText: "",
    quoteAuthor: ""
  },
  status: {
    isLoading: false,
    isError: false
  }
};
