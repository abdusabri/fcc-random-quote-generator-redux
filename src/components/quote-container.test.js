import mockedGetQuote from "../api/quotes-api";
import React from "react";
import { render, wait, fireEvent } from "@testing-library/react";
import pretty from "pretty";
import QuoteContainer from "./quote-container";
import { Provider } from "react-redux";
import store from "../store";

jest.mock("../api/quotes-api");

const fakeData = {
  quoteText: "This is a Quote",
  quoteAuthor: "Someone"
};

mockedGetQuote
  .mockImplementationOnce(async () => Promise.resolve(fakeData))
  .mockImplementationOnce(async () => Promise.resolve("error"));

describe("Quote Container", () => {
  it("Renders properly", async () => {
    const { getByText, container } = render(
      <Provider store={store}>
        <QuoteContainer />
      </Provider>
    );

    await wait();

    expect(getByText("This is a Quote")).toBeInTheDocument();
    expect(getByText(fakeData.quoteAuthor)).toBeInTheDocument();

    expect(pretty(container.innerHTML)).toMatchSnapshot();

    fireEvent.click(getByText(/Generate Quote/i));

    await wait();

    expect(getByText("Oops! Something is not right!")).toBeInTheDocument();

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
