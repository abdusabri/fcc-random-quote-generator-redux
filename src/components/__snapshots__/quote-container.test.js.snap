// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Container Renders properly 1`] = `
"<div class=\\"MuiPaper-root MuiCard-root QuoteCard-card-1 MuiPaper-elevation1 MuiPaper-rounded\\">
  <div class=\\"MuiCardHeader-root\\">
    <div class=\\"MuiCardHeader-content\\"><span class=\\"MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\"><div style=\\"display: flex; justify-items: left; justify-content: left;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"new-quote\\"><span class=\\"MuiButton-label\\">Generate Quote</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></span>
  </div>
</div>
<hr class=\\"MuiDivider-root\\">
<div class=\\"MuiCardContent-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\">
    <path d=\\"M10,7L8,11H11V17H5V11L7,7H10M18,7L16,11H19V17H13V11L15,7H18Z\\"></path>
  </svg><span id=\\"text\\">This is a Quote</span><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\">
    <path d=\\"M14,17H17L19,13V7H13V13H16M6,17H9L11,13V7H5V13H8L6,17Z\\"></path>
  </svg><br><span id=\\"author\\" style=\\"margin-top: 1em; display: block;\\">Someone</span></div>
<hr class=\\"MuiDivider-root\\">
<div class=\\"MuiCardActions-root MuiCardActions-spacing\\"><a id=\\"tweet-quote\\" href=\\"https://twitter.com/intent/tweet?text=&quot;This is a Quote&quot; by Someone\\" target=\\"blank\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Tweet Quote\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M22.46,6C21.69,6.35 20.86,6.58 20,6.69C20.88,6.16 21.56,5.32 21.88,4.31C21.05,4.81 20.13,5.16 19.16,5.36C18.37,4.5 17.26,4 16,4C13.65,4 11.73,5.92 11.73,8.29C11.73,8.63 11.77,8.96 11.84,9.27C8.28,9.09 5.11,7.38 3,4.79C2.63,5.42 2.42,6.16 2.42,6.94C2.42,8.43 3.17,9.75 4.33,10.5C3.62,10.5 2.96,10.3 2.38,10C2.38,10 2.38,10 2.38,10.03C2.38,12.11 3.86,13.85 5.82,14.24C5.46,14.34 5.08,14.39 4.69,14.39C4.42,14.39 4.15,14.36 3.89,14.31C4.43,16 6,17.26 7.89,17.29C6.43,18.45 4.58,19.13 2.56,19.13C2.22,19.13 1.88,19.11 1.54,19.07C3.44,20.29 5.7,21 8.12,21C16,21 20.33,14.46 20.33,8.79C20.33,8.6 20.33,8.42 20.32,8.23C21.16,7.63 21.88,6.87 22.46,6Z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></a></div>
</div>"
`;

exports[`Quote Container Renders properly 2`] = `
"<div class=\\"MuiPaper-root MuiCard-root QuoteCard-card-1 MuiPaper-elevation1 MuiPaper-rounded\\">
  <div class=\\"MuiCardHeader-root\\">
    <div class=\\"MuiCardHeader-content\\"><span class=\\"MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\"><div style=\\"display: flex; justify-items: left; justify-content: left;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"new-quote\\"><span class=\\"MuiButton-label\\">Generate Quote</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></span>
  </div>
</div>
<hr class=\\"MuiDivider-root\\">
<div class=\\"MuiCardContent-root\\">
  <div class=\\"MuiPaper-root QuoteCard-root-3 MuiPaper-elevation1 MuiPaper-rounded\\"><span>Oops! Something is not right!</span></div>
</div>
<hr class=\\"MuiDivider-root\\">
<div class=\\"MuiCardActions-root MuiCardActions-spacing\\"><a id=\\"tweet-quote\\" href=\\"https://twitter.com/intent/tweet?text=&quot;&quot; by \\" target=\\"blank\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Tweet Quote\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M22.46,6C21.69,6.35 20.86,6.58 20,6.69C20.88,6.16 21.56,5.32 21.88,4.31C21.05,4.81 20.13,5.16 19.16,5.36C18.37,4.5 17.26,4 16,4C13.65,4 11.73,5.92 11.73,8.29C11.73,8.63 11.77,8.96 11.84,9.27C8.28,9.09 5.11,7.38 3,4.79C2.63,5.42 2.42,6.16 2.42,6.94C2.42,8.43 3.17,9.75 4.33,10.5C3.62,10.5 2.96,10.3 2.38,10C2.38,10 2.38,10 2.38,10.03C2.38,12.11 3.86,13.85 5.82,14.24C5.46,14.34 5.08,14.39 4.69,14.39C4.42,14.39 4.15,14.36 3.89,14.31C4.43,16 6,17.26 7.89,17.29C6.43,18.45 4.58,19.13 2.56,19.13C2.22,19.13 1.88,19.11 1.54,19.07C3.44,20.29 5.7,21 8.12,21C16,21 20.33,14.46 20.33,8.79C20.33,8.6 20.33,8.42 20.32,8.23C21.16,7.63 21.88,6.87 22.46,6Z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></a></div>
</div>"
`;
